### ISSUE TITLE
AcadosOcpBatchSolver add more efficient interactions

### ISSUE THREAD

**Issue by itaouil** [acados/acados#1114]
Hi,

I recently started to use the AcadoBatchSolver in order to solve multiple OCPs in parallel using OpenMP.

The solving part works great, but when I need to populate several OCPs with different constraints and states, it takes a lot of time just to set the OCPs. Do you have any recommendations on how to speed up this part (maybe there is some ACADOS feature that I could use)?

Best,
Ilyass

**Comment by itaouil** [comment]
Hi @FreyJo,

I would be happy to try and contribute to this feature. Can you share some references or directions on how I could start with this feature?

Best,
Ilyass

**Comment by sandmaennchen** [comment]
Hi Ilyass,

two recent features that might speed up setting up the OCPs:

- global parameters: If your parameters are not changing across stages, you can use `p_global` instead of `p`. Global parameters also make use of precomputation , cf. `AcadosOcpSolver.set_p_global_and_precompute_dependencies()`.
 (This requires CasADi pre-relase, cf. https://github.com/acados/acados/pull/1234)
- flat setters and getters: `AcadosOcpSolver.set_flat()` and `AcadosOcpSolver.get_flat()` allow to set/get the iterate for the whole OCP. This is much faster than setting/getting stagewise.

Regarding the batch solver: I just added a batched version of `set_flat` to the `AcadosOcpBatchSolver` in this PR https://github.com/acados/acados/pull/1349. The changes should be rather minimal and might be a good start for you to further extend the batch solver.  Feel free to reach out if something's unclear! 

Best, Katrin

**Comment by sandmaennchen** [comment]
More functionality has been implemented in https://github.com/acados/acados/pull/1360.

**Comment by FreyJo** [comment]
This has become a very "generic" feature request. If more specific functionality is needed, please open a new issue.
